pkg/controller/dgraph/models/query/cluster.go

```diff
@@ -58,18 +57,23 @@ func RetrieveClusterHierarchy(view string) JSONDataWrapper {
 	return root
 }
 
+func getClusterMetricsQuery(view string) string {
+	switch view {
+	case Physical:
+		return getMetricsQueryForPhysicalResources()
+	case Logical:
+		return getMetricsQueryForLogicalResources()
+	default:
+		return ""
+	}
+}
+
 // RetrieveClusterMetrics returns all namespaces with metrics if view is logical and
 // returns all nodes and disks with metrics if view is physical
 func RetrieveClusterMetrics(view string) JSONDataWrapper {
-	var query string
-	if view == Physical {
-		query = getQueryForPhysicalResources()
-	} else {
-		query = getQueryForLogicalResources()
-	}
-
+	query := getClusterMetricsQuery(view)
 	parentRoot := ParentWrapper{}
-	err := dgraph.ExecuteQuery(query, &parentRoot)
+	err := executeQuery(query, &parentRoot)
 	calculateAggregateMetrics(&parentRoot)
```

Comment:
I'll suggest a rename here. 
Make the `getClusterMetricsQuery(view string)` a public function as `RetrieveClusterMetrics(view string)` and the other function as private as `executeQuery(view string) JSONDataWrapper` making the purpose of the functions more clear.

Reply:
Also, then have the executeQuery function common for both the hierarchy and metrics if it's the same.

Reply:
`executeQuery` is a variable set equal to `dgraph.ExecuteQuery`. This will allow us to mock dgraph execute query simply by changing `executeQuery` to a mock-dgraph-query-execution.